parquet-column/src/main/java/org/apache/parquet/internal/filter2/columnindex/RowRanges.java

```diff
@@ -299,6 +300,10 @@ public boolean isOverlapping(long from, long to) {
         (r1, r2) -> r1.isBefore(r2) ? -1 : r1.isAfter(r2) ? 1 : 0) >= 0;
   }
 
+  public List<Range> getRanges() {
+    return ranges;
```

Comment:
I don't like returning `ranges` directly. This is a mutable list so the caller can modify it without any control. I would suggest wrapping it into an unmodifiable list.

Reply:
Yeah, that makes sense. After this PR, I will change the master branch also.